Classes/Controller/ModuleController.php

```diff
@@ -114,14 +122,17 @@ public function newAction(User $user = null): void
      * @param array $password Expects an array in the format array('<password>', '<password confirmation>')
      * @param User $user The user to create
      * @param \DateTime $expirationDate
+     * @param array $roleIdentifiers Identifiers of roles to assign to account
      * @return void
      * @Flow\Validate(argumentName="username", type="\Neos\Flow\Validation\Validator\NotEmptyValidator")
      * @Flow\Validate(argumentName="username", type="\Neos\Neos\Validation\Validator\UserDoesNotExistValidator")
      * @Flow\Validate(argumentName="password", type="\Neos\Neos\Validation\Validator\PasswordValidator", options={ "allowEmpty"=0, "minimum"=1, "maximum"=255 })
      */
-    public function createAction($username, array $password, User $user, ?\DateTime $expirationDate): void
+    public function createAction($username, array $password, User $user, ?\DateTime $expirationDate, array $roleIdentifiers = []): void
     {
-        $user = $this->userService->addUser($username, $password[0], $user, [self::$roleIdentifier], self::$authenticationProviderName);
+        // make sure self::$roleIdentifier is always added
```

Comment:
possible privilege escalation? i.e. are the given $roleIdentifiers checked against allowed roles?

Reply:
Now only "FE user roles" can be assigned.